doc/api/tls.md

```diff
@@ -334,6 +334,34 @@ added: v0.3.2
 The `tls.Server` class is a subclass of `net.Server` that accepts encrypted
 connections using TLS or SSL.
 
+### Event: 'keylog'
+<!-- YAML
+added: REPLACEME
+-->
+
+* `line` {Buffer} Line of ASCII text, in NSS `SSLKEYLOGFILE` format.
+* `tlsSocket` {tls.TLSSocket} The `tls.TLSSocket` instance on which it was
+  generated.
+
+The `keylog` event is emitted when key material is generated or received by
+a connection to this server (typically before handshake has completed, but not
+necessarily). This keying material can be stored for debugging, as it allows
+captured TLS traffic to be decrypted. It may be emitted multiple times for
+each socket.
+
+A typical use case is to append received lines to a common text file, which
+is later used by software (such as Wireshark) to decrypt the traffic:
+
+```js
+const logFile = fs.createWriteStream('/tmp/ssl-keys.log', { flags: 'a' });
+// ...
+server.on('keylog', (line, tlsSocket) => {
+  if (tlsSocket.remoteAddress !== '...')
+    return; // Only log keys for a particular IP
+  logFile.write(`${line}\n`);
```

Comment:
Since people tend to copy/paste from the docs I would suggest writing this as:
```js
logFile.write(line);
logFile.write('\n');
```
(Ditto line 673.)

Alternatively, you could append the newline when you create the buffer:
```c++
const size_t size = strlen(line);
Local<Value> line_bf = Buffer::Copy(env, line, 1 + size).ToLocalChecked();
char* data = Buffer::Data(line_bf);
data[size] = '\n';
```
It's not super elegant but that way people don't have to think about it (unless they don't want the newline, of course.)

Reply:
As for the first suggestion, couldn't this cause log corruption if multiple programs are writing to the log (as is often the case)? I thought it had to be on a single write to be atomic.

As for the second, some people might want to parse the secrets or store them in a different medium or with i.e. different line terminators, I'm not keen on adding the newline :thinking:

Reply:
Its just a suggestion, you don't have to do it, but I like the idea of making the data NL terminated. It resolves the conflict that you point out between wanting atomic writes of a single line to avoid file corruption, and wanting to not take every Buffer and turn it into a String. If what is usually wanted is a single Buffer with a newline that can be appended to a file, then why not emit that directly? It makes the most common use-case more ergonomic, but does not make parsing harder, or make it harder to do any other transformations a person might want (like stripping/replacing newlines).

Reply:
Hmm, that's true... adding the terminator then (sorry about all those CI runs :sweat_smile:)

Reply:
> As for the first suggestion, couldn't this cause log corruption if multiple programs are writing to the log (as is often the case)?

That's a good point. I hadn't considered that.

> I thought it had to be on a single write to be atomic.

You're right, but being a single write doesn't guarantee it's atomic per se. [This blog post](https://nullprogram.com/blog/2016/08/03/) has some good discussion.